### PR TITLE
Fix server heartbeat logging error after deleting server.#7540

### PR DESCRIPTION
**Pull request by RohitBhati8269**
Detail Info:- https://github.com/pgadmin-org/pgadmin4/issues/7540

### CODE REVIEW

File: web/pgadmin/browser/static/js/heartbeat.js [hunk 1]
```diff
@@ -28,6 +28,8 @@ export function send_heartbeat(_server_id, _item) {
       .catch((error) => {
         if (error && error.message == 'Network Error') {
           pgAdmin.Browser.notifier.error(gettext(`pgAdmin server not responding, try to login again: ${error.message || error.response.data.errormsg}`));
+        } else if (error.response.data?.data?.includes('Manager not found')) {
+          console.error(error);
```
Comment (khushboovashi) [inline web/pgadmin/browser/static/js/heartbeat.js:32]: No need for a console error, just bypass the error.

File: web/pgadmin/utils/heartbeat.py [hunk 2]
```diff
@@ -25,22 +25,28 @@ def log_server_heartbeat(data):
...
-    if session.sid not in _server_heartbeat:
-        _server_heartbeat[session.sid] = {}
+    if session_id not in _server_heartbeat:
+        _server_heartbeat[session_id] = {}
 
     if not manager:
         stop_server_heartbeat(data)
-        return False, gettext("Manager not found. Stopped Heartbeat logging.")
+        msg = gettext("Manager not found. Stopped Heartbeat logging.")
+        current_app.logger.debug(
```
Comment (khushboovashi) [inline web/pgadmin/utils/heartbeat.py:36]: It should be current_app.logger.error

**Review by khushboovashi — CHANGES_REQUESTED** [review]

**Review by khushboovashi — CHANGES_REQUESTED** [review]
It showing log error in some cases.
Screenshot 2024-08-06 at 4 06 15 PM

File: web/pgadmin/browser/static/js/heartbeat.js [hunk 3]
```diff
@@ -26,10 +26,12 @@ export function send_heartbeat(_server_id, _item) {
         }
       })
       .catch((error) => {
-        if (error && error.message == 'Network Error') {
-          pgAdmin.Browser.notifier.error(gettext(`pgAdmin server not responding, try to login again: ${error.message || error.response.data.errormsg}`));
-        } else {
-          pgAdmin.Browser.notifier.error(gettext(`Server heartbeat logging error: ${error.message || error.response.data.errormsg}`));
+        if (!error.response?.data?.data?.includes('Manager not found')){
```
Comment (khushboovashi) [inline web/pgadmin/browser/static/js/heartbeat.js:29]: I would just return from this if condition, rather than putting everything into this condition.

**Review by khushboovashi — CHANGES_REQUESTED** [review]

File: web/pgadmin/browser/static/js/heartbeat.js (hunk 3 above)
Comment (RohitBhati8269) [reply web/pgadmin/browser/static/js/heartbeat.js:29]: If I directly return  then it will not call the  `stop_heartbeat(_item);`

**Pull request merged**